toolkit.py

```diff
@@ -0,0 +1,196 @@
+        return self.get_board_boundaries(process_image)
+
+    def __get_file_name(self, path):
+        """
+        Args:
+            path(string): Path to file
+        Returns:
+            string: File name extracted from path
+        """
+        head, tail = ntpath.split(path)
+        name = tail or ntpath.basename(head)
+        return name.split('.')[0]
+
+    def write_video_metadata(self, video, meta_file):
+        """
+        Sets up a video capture and loops through all of the
+        frames. Pulls frame metadata from every frame and stores
+        it into an array. After we are done with a video we flush these
+        values to a file
+
+        Args:
+            video(string): Name of video file to load
+            meta_file(string): Name of file where metadata will be stored
+        """
+        vid = cv.VideoCapture(video)
+        if vid is None:
+            print("Couldn't load video {}".format(video), file=sys.stderr)
+            return
+        frames = []
+        while(vid.isOpened()):
+            ret, frame = vid.read()
+            if not ret:
+                break
+            (x, y) = self.get_table_metadata(frame)
+            frames.append((x, y))
+        with meta_file as csvfile:
+            for i, frame in enumerate(frames):
+                csv_writer = csv.writer(csvfile, lineterminator='\n')
+                csv_writer.writerow([x[0], x[1], y[0], y[1]])
+        vid.release()
+
+    def write_image_metadata(self, image, meta_file):
+        """
+        Reads given image and gets his metadata which
+        are written to a file
+
+        Args:
+            image(string): Name of image to load
+            meta_file(string): Name of file where metadata will be stored
+        """
+        im = cv.imread(image, cv.CV_LOAD_IMAGE_COLOR)
+        if im is None:
+            print("Couldn't read image file {}".format(image), file=sys.stderr)
+            return
+        (x, y) = self.get_table_metadata(im)
+        with meta_file as csvfile:
+            csv_writer = csv.writer(csvfile, lineterminator='\n')
+            csv_writer.writerow([x[0], x[1], y[0], y[1]])
+
+    video_extension_list = ("mkv", "wmv", "avi", "mp4")
+
+    @timeit
+    def write_metadata(self, input_file, output_file=None):
+        """
+        Main function for getting metadata from images or videos.
+        It checks if format of a file is supported and then calls
+        approperiate function
+
+        Args:
+            input_file(string): Path to file to process
+            output_file(string): Path to output file where metadata
+                                 will be stored
+        """
+        if(output_file is None):
+            file_name = self.__get_file_name(input_file)
+            output_file = open("{}_meta.csv".format(file_name), 'wb+')
+
+        if(input_file.endswith(self.video_extension_list)):
+            self.write_video_metadata(input_file, output_file)
+        elif (imghdr.what(input_file)):
```

Comment:
@Foxiy While I understand that `imghdr.what` returns a NoneType when the file is not recognized, figuring that out required me to read the docs and trying to dissolve this magic into some sort of sensible pieces of information.

I believe it would be better to say `imghdr.what(input_file) is not None)`, just for the sake of clarity.